src/OpenFeature.Contrib.Providers.GOFeatureFlag/evaluator/IEvaluator.cs

```diff
@@ -0,0 +1,78 @@
+using System.Threading.Tasks;
+using OpenFeature.Model;
+
+namespace OpenFeature.Contrib.Providers.GOFeatureFlag.evaluator;
+
+/// <summary>
+///     IEvaluator is an interface that represents the evaluation of a feature flag.
+///     It can have multiple implementations: Remote or InProcess.
+/// </summary>
+public interface IEvaluator
+{
+    /// <summary>
+    ///     Initialize the evaluator.
+    /// </summary>
+    public Task InitializeAsync();
+
+    /// <summary>
+    ///     Shutdown the evaluator.
+    /// </summary>
+    public Task DisposeAsync();
```

Comment:
Since `DisposeAsync()` is part of the interface, it would be cleaner to inherit from [IAsyncDisposable](https://learn.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-disposeasync). This enables proper usage of `await using` and aligns with standard .NET patterns.